src/converter/jfr2pprof.java

```diff
@@ -116,26 +133,45 @@ public void dump(final OutputStream out) throws Exception {
 
         final Proto sampleType = new Proto(100);
 
-        profile.field(PROFILE_STRING_TABLE, "cpu".getBytes(StandardCharsets.UTF_8));
-        sampleType.field(VALUETYPE_TYPE, stringId++);
+        if (eventClass == AllocationSample.class) {
+            profile.field(PROFILE_STRING_TABLE, "alloc_space".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_TYPE, stringId++);
+            profile.field(PROFILE_STRING_TABLE, "nanoseconds".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_UNIT, stringId++);
+        } else {
+            profile.field(PROFILE_STRING_TABLE, "cpu".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_TYPE, stringId++);
+            profile.field(PROFILE_STRING_TABLE, "nanoseconds".getBytes(StandardCharsets.UTF_8));
+            sampleType.field(VALUETYPE_UNIT, stringId++);
+        }
 
-        profile.field(PROFILE_STRING_TABLE, "nanoseconds".getBytes(StandardCharsets.UTF_8));
-        sampleType.field(VALUETYPE_UNIT, stringId++);
+        profile.field(PROFILE_STRING_TABLE, "allocation_size".getBytes(StandardCharsets.UTF_8));
+        final int allocationSizeKeyId = stringId++;
+        profile.field(PROFILE_STRING_TABLE, "bytes".getBytes(StandardCharsets.UTF_8));
+        final int allocationSizeUnitId = stringId++;
```

Comment:
Are these strings only need for allocation profile? Does it make sense to put them under `if (eventClass == AllocationSample.class)`?